frontend/src/scenes/data-warehouse/editor/OutputPaneTabs/lineageTabLogic.ts

```diff
@@ -0,0 +1,155 @@
+import { actions, connect, events, kea, key, listeners, path, props, reducers, selectors } from 'kea'
+import { subscriptions } from 'kea-subscriptions'
+import api from 'lib/api'
+import { databaseTableListLogic } from 'scenes/data-management/database/databaseTableListLogic'
+import { Node } from 'scenes/data-model/types'
+import { dataWarehouseViewsLogic } from 'scenes/data-warehouse/saved_queries/dataWarehouseViewsLogic'
+
+import { DataWarehouseSavedQuery } from '~/types'
+
+import { multitabEditorLogic } from '../multitabEditorLogic'
+import type { lineageTabLogicType } from './lineageTabLogicType'
+
+export interface LineageTabLogicProps {
+    codeEditorKey: string
+}
+
+export const lineageTabLogic = kea<lineageTabLogicType>([
+    path(['data-warehouse', 'editor', 'outputPaneTabs', 'lineageTabLogic']),
+    props({} as LineageTabLogicProps),
+    key((props) => props.codeEditorKey),
+    actions({
+        loadNodes: true,
+        traverseAncestors: (viewId: DataWarehouseSavedQuery['id'], level: number) => ({ viewId, level }),
+        setNodes: (nodes: Record<string, Node>) => ({ nodes }),
+    }),
+    connect((props: LineageTabLogicProps) => ({
+        values: [
+            multitabEditorLogic({ key: props.codeEditorKey }),
+            ['metadata'],
+            databaseTableListLogic,
+            ['posthogTablesMap', 'viewsMapById', 'dataWarehouseTablesMapById'],
+            dataWarehouseViewsLogic,
+            ['dataWarehouseSavedQueryMap'],
+        ],
+        actions: [
+            multitabEditorLogic({ key: props.codeEditorKey }),
+            ['runQuery'],
+            dataWarehouseViewsLogic,
+            ['loadDataWarehouseSavedQueries'],
+        ],
+    })),
+    reducers({
+        nodeMap: [
+            {} as Record<string, Node>,
+            {
+                setNodes: (_, { nodes }) => nodes,
+            },
+        ],
+    }),
+    listeners(({ actions, values }) => ({
+        loadNodes: async () => {
+            const nodes: Record<string, Node> = {}
+
+            const traverseAncestors = async (viewId: DataWarehouseSavedQuery['id'], level: number): Promise<void> => {
+                if (!nodes[viewId]?.savedQueryId) {
+                    return
+                }
+
+                const result = await api.dataWarehouseSavedQueries.ancestors(viewId, level)
+                for (const ancestor of result.ancestors) {
+                    nodes[ancestor] = {
+                        nodeId: ancestor,
+                        name:
+                            values.viewsMapById[ancestor]?.name ||
+                            values.dataWarehouseTablesMapById[ancestor]?.name ||
+                            ancestor,
+                        savedQueryId: values.viewsMapById[ancestor]?.id,
+                        leaf: [...(nodes[ancestor]?.leaf || []), viewId],
+                    }
+                    await traverseAncestors(ancestor, 1)
+                }
+            }
```

Comment:
I imagine this could create a lot of API requests if we have a large number of dependents on a node/view - maybe something to rethink in the future